src/Swift.Bindings/src/Parser/ModuleProcessor.cs

```diff
@@ -0,0 +1,223 @@
+using Swift.Runtime;
+
+namespace BindingsGeneration
+{
+    /// <summary>
+    /// Performs post-processing of types collected from the Swift ABI before generating bindings.
+    /// Calculates properties on types which are not directly available from the ABI.
+    /// Generates type database entries for structs, enums, and classes.
+    /// </summary>
+    internal class ModuleProcessor
```

Comment:
I like the idea of having a processor/marshaller that based on ABI and dylib inputs, registers types in the database. For that, we need to wire up a Mach-O reader and a demangler.

Reply:
Lets address this in a follow up: https://github.com/dotnet/runtimelab/issues/2918